src/Core/Content/Product/SalesChannel/Price/ProductPriceDefinitionBuilder.php

```diff
@@ -18,33 +18,45 @@ class ProductPriceDefinitionBuilder implements ProductPriceDefinitionBuilderInte
 {
     public function build(ProductEntity $product, SalesChannelContext $context, int $quantity = 1): ProductPriceDefinitions
     {
-        $listingPrice = $this->buildListingPriceDefinition($product, $context);
+        $matchingRulePrices = $this->getFirstMatchingPriceRule($product->getPrices(), $context);
+        $listingPrice = $this->buildListingPriceDefinition($product, $matchingRulePrices, $context);
+        $quantity = $this->getMinPurchase($product, $quantity, $matchingRulePrices);
 
         return new ProductPriceDefinitions(
-            $this->buildPriceDefinition($product, $context),
-            $this->buildPriceDefinitions($product, $context),
+            $this->buildPriceDefinition($product, $matchingRulePrices, $context),
+            $this->buildPriceDefinitions($product, $matchingRulePrices, $context),
             $listingPrice['from'],
             $listingPrice['to'],
-            $this->buildPriceDefinitionForQuantity($product, $context, $quantity)
+            $this->buildPriceDefinitionForQuantity($product, $matchingRulePrices, $context, $quantity)
         );
     }
 
-    private function buildPriceDefinitions(ProductEntity $product, SalesChannelContext $context): PriceDefinitionCollection
+    /**
+     * @param ProductPriceEntity[]|null $matchingRulePrices
+     */
+    private function buildPriceDefinitions(ProductEntity $product, ?array $matchingRulePrices, SalesChannelContext $context): PriceDefinitionCollection
     {
-        $taxRules = $context->buildTaxRules($product->getTaxId());
-
-        $prices = $this->getFirstMatchingPriceRule($product->getPrices(), $context);
-
-        if (!$prices) {
+        if (!$matchingRulePrices) {
```

Comment:
please use explicit cheeks like `=== null`

Reply:
also state what should happen with an empty array

Reply:
Hey @mitelg,
do you have any background for me? Didn't see any need for this because this `inexplicit` check will catch both 🤔 
http://sandbox.onlinephpfunctions.com/code/8cb133ea74c092a3fbe2315981c40d249dbe36a0